followthemoney/export/csv.py

```diff
@@ -1,26 +1,38 @@
 import csv
 from pathlib import Path
 from banal import ensure_list
+from typing import Union, Optional, Dict, Tuple, Any, TextIO, Iterable
 
 from followthemoney.export.common import Exporter
+from followthemoney.schema import Schema
+from followthemoney.proxy import EntityProxy
 
 
-class CSVMixin(object):
+class CSVMixin(Exporter):
 
-    def _configure(self, directory, dialect=csv.unix_dialect, extra=None):
+    def _configure(self, directory, dialect=csv.unix_dialect,
+                   extra: Optional[str]=None):
         self.directory = Path(directory)
         self.extra = ensure_list(extra)
         self.dialect = dialect
-        self.handles = {}
+        self.handles: Dict[Schema, Tuple[TextIO, Any]] = {}
 
-    def _open_csv_file(self, name):
+    def _open_csv_file(self, name) -> Tuple[TextIO, Any]:
         self.directory.mkdir(parents=True, exist_ok=True)
         file_path = self.directory.joinpath('{0}.csv'.format(name))
         handle = open(file_path, mode='w')
         writer = csv.writer(handle, dialect=self.dialect)
         return handle, writer
 
-    def _get_writer(self, schema):
+    def _write_header(self, writer, schema: Schema):
```

Comment:
Does this apply 1:1 for the Neo4J Bulk exporter, the other user of the CSVMixIn?

Reply:
.... I realized that I missed adding typing in the neo4j exporter, so I'm pretty sure it doesn't vaguely work. I'll look at that.

Reply:
So, I checked, yes, it will works. the method `_write_header` moved from `CSVExporter` to `CSVMixin` is overwritten in `Neo4JCSVExporter`